app/models/csv_export.rb

```diff
@@ -0,0 +1,39 @@
+class CsvExport < ActiveRecord::Base
+  belongs_to :user
+  STATUS_VALUES = ['pending', 'started', 'finished', 'downloaded', 'failed', 'deleted']
+
+  before_destroy :file_delete
+
+  def status?(state)
+    'ready' == state.to_s ? ['downloaded', 'finished'].include?(status) : status == state.to_s
+  end
+
+  def email
+    user.try :email
```

Comment:
can replace this method by something like `delegate :email, to: :user, allow_nil: true`